pal/image/base-image-data.ts

```diff
@@ -90,6 +92,49 @@ export class BaseImageData {
         }
     }
 
+    public getRawData (): RawDataType | null {
+        if (this._source == null) {
+            return null;
+        }
+        let data: ArrayBufferView | null = null;
+        if ('_data' in this._source) {
+            data = this._source._data;
+        } else if ('getContext' in this._source) {
+            const canvasElem = this._source;
+            const imageData = canvasElem.getContext('2d')?.getImageData(0, 0, this._source.width, this._source.height);
+            const buff = imageData!.data.buffer;
+            let rawBuffer;
+            if ('buffer' in buff) {
+                // es-lint as any
+                data = new Uint8Array((buff as any).buffer, (buff as any).byteOffset, (buff as any).byteLength);
+            } else {
+                rawBuffer = buff;
+                data = new Uint8Array(rawBuffer);
+            }
+        } else if (this._source instanceof HTMLImageElement || this._source instanceof ImageBitmap) {
+            const img = this._source;
+            const canvas = ccwindow.document.createElement('canvas');
+            canvas.width = img.width;
+            canvas.height = img.height;
+            const ctx = canvas.getContext('2d');
+            ctx?.drawImage(img as any, 0, 0);
+            const imageData = ctx?.getImageData(0, 0, img.width, img.height);
+            const buff = imageData!.data.buffer;
+            let rawBuffer;
+            if ('buffer' in buff) {
+                // es-lint as any
+                data = new Uint8Array((buff as any).buffer, (buff as any).byteOffset, (buff as any).byteLength);
+            } else {
+                rawBuffer = buff;
+                data = new Uint8Array(rawBuffer);
+            }
+        } else {
+            // eslint-disable-next-line no-console
+            console.log('imageBmp copy not impled!');
```

Comment:
impled -> implemented

`console.warn('getRawData doesn't support this source type.')`

Reply:
Yep, it would be better to use assert here.